pymeasure/instruments/attocube/anc300.py

```diff
@@ -255,3 +299,76 @@ def stop_all(self):
             attribute = getattr(self, attr)
             if isinstance(attribute, Axis):
                 attribute.stop()
+
+    def handle_deprecated_host_arg(self, adapter, kwargs):
+        """
+        This function formats user input to the __init__ function to be compatible with the
+        current definition of the __init__ function. This is used to support outdated (deprecated)
+        code. and separated out to make it easier to remove in the future. To whoever removes this:
+        This function should be removed and the `adapter` argument in the __init__ method should
+        be made non-optional.
+
+        :param dict kwargs: keyword arguments passed to the __init__ function,
+            including the deprecated `host` argument.
+        :return str: resource string for the VISAAdapter
+        """
+        host = kwargs.pop("host", None)
```

Comment:
If I see it correctly then the `host` kwarg did not exist before this PR. It was a positional argument. Can one not remove the `host` kwarg? I think applying the magic below on the adapter argument alone would be sufficient?

Reply:
This was added so that if someone had ANC300Controller(`host=168.169.0.2`, ...) was given. Then requires no update for previous code to continue working without changes.

> I am not sure if the host keyword argument you show with the deprecation warning makes a lot of sense. Previously host was a regular argument so the chance that someones old code works without adapting are very small.

This now works both if previously someone code had ANC300Controller(`host=168.169.0.2`, ...)  or ANC300Controller(`168.169.0.2`, ...). What am I missing here?

Reply:
ok, you are right, this is the most backward compatible solution conceivable.